openhantek/src/settings.h

```diff
@@ -23,7 +23,7 @@ struct DsoSettingsOptions {
 /// \brief Holds the settings of the program.
 class DsoSettings {
   public:
-    DsoSettings();
+    explicit DsoSettings(unsigned int channels);
     bool setFilename(const QString &filename);
 
     void setChannelCount(unsigned int channels);
```

Comment:
A DSO will never change it's channels during runtime I guess. We should remove this method all together or make it private. The logic within `setChannelCount` is highly complex because of this non existing use-case.

Reply:
Sure, removal of unused channels in setChannelCount() looks weird, so I am removing the method. Remaining initialization will go directly to DsoSettings ctor.